pytrip/ctx.py

```diff
@@ -70,39 +72,49 @@ def create_dicom(self):
+        ds.HighBit = self.num_bytes * 8 - 1
+        ds.PatientPosition = 'HFS'
+        ds.RescaleIntercept = 0.0
+        ds.ImageType = ['ORIGINAL', 'PRIMARY', 'AXIAL']
+        ds.PatientPosition = 'HFS'
+        # ds.SeriesInstanceUID is created in the top-level cube class
+        # ds.SeriesInstanceUID = '2.16.840.1.113662.2.12.0.3057.1241703565.43'
+        ds.RescaleSlope = 1.0
+        ds.PixelRepresentation = 1
+        ds.SOPClassUID = '1.2.840.10008.5.1.4.1.1.2'  # CT Image Storage SOP Class
+
+        # Eclipse tags
+        from dicom.tag import Tag
+        ds.add_new(Tag(0x0018, 0x0060), 'DS', '')  # KVP
+        ds.add_new(Tag(0x0020, 0x0012), 'IS', '')  # AcquisitionNumber
+
         for i in range(len(self.cube)):
-            ds = self.create_dicom_base()
-            ds.Modality = 'CT'
-            ds.SamplesperPixel = 1
-            ds.BitsAllocated = self.num_bytes * 8
-            ds.BitsStored = self.num_bytes * 8
-            ds.HighBit = self.num_bytes * 8 - 1
-            ds.PatientPosition = 'HFS'
-            ds.RescaleIntercept = 0.0
-            ds.ImageType = ['ORIGINAL', 'PRIMARY', 'AXIAL']
-
-            ds.PatientPosition = 'HFS'
-            ds.SeriesInstanceUID = '2.16.840.1.113662.2.12.0.3057.1241703565.43'
-            ds.RescaleSlope = 1.0
-            ds.PixelRepresentation = 1
-            ds.ImagePositionPatient = ["%.3f" % (self.xoffset * self.pixel_size),
-                                       "%.3f" % (self.yoffset * self.pixel_size),
-                                       "%.3f" % (self.slice_pos[i])]
-            ds.SOPClassUID = '1.2.840.10008.5.1.4.1.1.2'  # CT Image Storage SOP Class
-            ds.SOPInstanceUID = '2.16.1.113662.2.12.0.3057.1241703565.' + str(i + 1)
-
-            ds.SeriesDate = '19010101'  # !!!!!!!!
-            ds.ContentDate = '19010101'  # !!!!!!
-            ds.SeriesTime = '000000'  # !!!!!!!!!
-            ds.ContentTime = '000000'  # !!!!!!!!!
-
-            ds.SliceLocation = str(self.slice_pos[i])
-            ds.InstanceNumber = str(i + 1)
-            pixel_array = np.zeros((ds.Rows, ds.Columns), dtype=self.pydata_type)
+            _ds = copy.deepcopy(ds)
+            _ds.ImagePositionPatient = ["%.3f" % (self.xoffset * self.pixel_size),
+                                        "%.3f" % (self.yoffset * self.pixel_size),
+                                        "%.3f" % (self.slice_pos[i])]
+
+            _ds.SOPInstanceUID = '2.16.1.113662.2.12.0.3057.1241703565.' + str(i + 1)
+
+            # .HED files do not carry any time stamp (other than the usual file meta data)
+            # so let's just fill it with current times. (Can be overriden by user)
+            _ds.SeriesDate = datetime.datetime.today().strftime('%Y%m%d')
+            _ds.ContentDate = datetime.datetime.today().strftime('%Y%m%d')
+            _ds.SeriesTime = datetime.datetime.today().strftime('%H%M%S')
+            _ds.ContentTime = datetime.datetime.today().strftime('%H%M%S')
```

Comment:
I think it is safer to assign the same generation time and date to all slices.

In 99% of cases they will anyway have the same time as the execution time of one loop is probably ~ms.

But in some cases it may happen that second is incremented in the clock and part of slices will get a timestamp i.e. 07:15:15 and rest of slices 07:15:16